crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -39,6 +42,8 @@ pub struct NameServerPool<P: ConnectionProvider + Send + 'static> {
     options: ResolverOpts,
 }
 
+static NEXT_INDEX: AtomicUsize = AtomicUsize::new(0);
```

Comment:
I don't think this should be a process-wide index, it should probably live inside the `NameServerPool`.